### PR TITLE
Changed syntax to use ES6 class and package prop-types since the are both deprecated

### PR DESCRIPTION
**Pull request by wind85**

### CODE REVIEW

**Comment by coremi** [comment]
you use two different ways to define properties inside of the class (displayName and propTypes) and both will, as far as I know, not work out of the box without installing another babel plugin. You could define the displayName using the constructor. The static propTypes are best defined after the class definition like so `FontAwesome.propTypes = {...}`

**Comment by wind85** [comment]
I followed your advice...

File: src/index.js [hunk 1]
```diff
@@ -22,26 +23,11 @@ import srOnlyStyle from './screen-reader-styles'
...
-    pulse: React.PropTypes.bool,
-    rotate: React.PropTypes.oneOf([ 90, 180, 270 ]),
-    size: React.PropTypes.oneOf([ 'lg', '2x', '3x', '4x', '5x' ]),
-    spin: React.PropTypes.bool,
-    stack: React.PropTypes.oneOf([ '1x', '2x' ]),
-    tag: React.PropTypes.string,
-  },
+class FontAwesome extends React.Component {
+  constructor() {
+    super();
```
Comment (danawoodman) [inline src/index.js]: Can you remove semicolon to stay consistent?

File: src/index.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import React from 'react'
+import PropTypes from 'prop-types';
```
Comment (danawoodman) [inline src/index.js]: Remove semi

File: src/index.js [hunk 3]
```diff
@@ -96,5 +82,27 @@ export default React.createClass({
...
+  pulse: PropTypes.bool,
+  rotate: PropTypes.oneOf([ 90, 180, 270 ]),
+  size: PropTypes.oneOf([ 'lg', '2x', '3x', '4x', '5x' ]),
+  spin: PropTypes.bool,
+  stack: PropTypes.oneOf([ '1x', '2x' ]),
+  tag: PropTypes.string,
+}
+
+
+export default FontAwesome;
```
Comment (danawoodman) [inline src/index.js]: Remove semi

**Comment by danawoodman** [comment]
@wind85 thanks for the PR! @coremi thanks for offering your feedback! Looks good minus some minor feedback!

**Comment by danawoodman** [comment]
Also, you're going to need to commit the built files by running `npm run dist` and committing the changes

**Comment by wind85** [comment]
Hi, ok tomorrow morning I will also sent you the files built by `npm run dist` ... Do you mind if I also update the readme as well, in particular the contributing section to reflect the need to commit the files build by `npm run dist` ...

**Comment by danawoodman** [comment]
@wind85 yes that would be great! Thanks!

File: examples/webpack.config.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-const webpack = require('webpack')
+// const webpack = require('webpack') unused variable
```
Comment (danawoodman) [inline examples/webpack.config.js]: Go ahead and remove this, thanks!

File: examples/index.js [hunk 5]
```diff
@@ -4,15 +4,15 @@ const FA = require('react-fontawesome')
...
   document.getElementById('spin')
 )
 
 // Stack
 ReactDOM.render(
-  <span className='fa-stack fa-lg'>
-    <FA name='square-o' stack='2x' />
-    <FA name='twitter' stack='1x' />
+  <span className="fa-stack fa-lg">
+    <FA name="square-o" stack="2x" />
```
Comment (danawoodman) [inline examples/index.js]: Please keep consistent with the styling of the app an use single quotes everywhere, thanks!

File: package.json [hunk 6]
```diff
@@ -75,6 +73,8 @@
     "react": ">=0.12.0"
   },
   "dependencies": {
+    "babel-cli": "^6.24.1",
```
Comment (danawoodman) [inline package.json]: Why were these moved here? They shouldn't be in dependencies since we build the application to not require any dependencies other than React

File: src/index.js [hunk 7]
```diff
@@ -45,8 +45,7 @@ class FontAwesome extends React.Component {
       stack,
       tag = 'span',
       ariaLabel,
-      ...props,
-    } = this.props
+      ...props } = this.props
```
Comment (danawoodman) [inline src/index.js]: Can you revert this change?

File: src/index.js [hunk 8]
```diff
@@ -1,5 +1,5 @@
 import React from 'react'
-import PropTypes from 'prop-types';
+import PropTypes from 'prop-types'
```
Comment (danawoodman) [inline src/index.js]: I just merged #40 which solves this issue

**Comment by danawoodman** [comment]
Thanks @wind85 for the PR! Just a few minor change requests before I merge! 🍻

**Comment by danawoodman** [comment]
@wind85 also, looks like you'll have to rebase because of the merging of #40

File: src/index.js (hunk 7 above)
Comment (wind85) [inline src/index.js]: About this change, it's a little problematic, since it brakes the linting, apparently it should not have a trailing comma but your linting configuration enforces it, that's a work-around in order to keep the linter happy...

File: examples/index.js (hunk 5 above)
Comment (wind85) [inline examples/index.js]: I changed those since that's the style enforced by your current linter configuration. Let me know how you want to fix it.

File: package.json (hunk 6 above)
Comment (wind85) [inline package.json]: My bad...

File: src/index.js (hunk 7 above)
Comment (danawoodman) [inline src/index.js]: Hmm, good to know, I will fix the linting rule later, for now it's ok to ignore that error

File: package.json (hunk 6 above)
Comment (danawoodman) [inline package.json]: No problem!

File: examples/index.js (hunk 5 above)
Comment (danawoodman) [inline examples/index.js]: Lets use single space and Ill fix the linter, sorry for that!

File: src/index.js (hunk 7 above)
Comment (wind85) [inline src/index.js]: Ok, fyi I did a little googling and apparently the spec says that in this case the the trailing comma should not be enforced. Anyway I will revert the change to the original state.

File: examples/index.js (hunk 5 above)
Comment (wind85) [inline examples/index.js]: Ok I will revert the changes to use the single quotes...

Comment (danawoodman) [inline examples/index.js]: Thanks!

File: src/index.js (hunk 7 above)
Comment (danawoodman) [inline src/index.js]: Thanks!

**Comment by danawoodman** [comment]
@wind85 once you rebase we can merge this in!

**Comment by wind85** [comment]
I am a bit of a git newbie, therefore I am not too familiar with the all rebase process. I added one more commit the fixes the last one, I am sorry about that.

**Comment by danawoodman** [comment]
@wind85 if you can make me a contributor to your fork I can do it for you 😄

**Comment by wind85** [comment]
Honestly that's my first contribution, I just started to use git some time ago. Is that ok, or I must rebase?

**Comment by danawoodman** [comment]
@wind85 yeah, we have to rebase since the other pull request changed some files. If you make me a contributor I can do it on your branch for you 👍

**Comment by danawoodman** [comment]
No worries about being new to Git/Github, I was too once! Happy to help you get this merged 😄

**Comment by wind85** [comment]
Alright then I will do it tomorrow...

**Comment by R1ZZU** [comment]
@danawoodman Hi! What is current status of this PR? Really looking forward to it 😏

**Comment by danawoodman** [comment]
@wind85 I need you to run `npm run dist` to build the compiled code. Once that is done, I can merge :)

cc @R1ZZU

**Comment by danawoodman** [comment]
@wind85 thanks again for your patience on this, I promise we will get this in soon!

**Comment by wind85** [comment]
@danawoodman Hi, you need to change the lint configuration otherwise it doesn't build, with the current settings...

**Comment by danawoodman** [comment]
You can try just `npm run build` for now

**Comment by wind85** [comment]
Anything wrong with it?

**Comment by danawoodman** [comment]
Thanks! Will check this out tomorrow!

**Comment by danawoodman** [comment]
Thanks @wind85! This is now in release `v1.6.1`!
